edit-post/index.js

```diff
@@ -47,13 +50,15 @@ export function reinitializeEditor( target, settings ) {
  * The return value of this function is not necessary if we change where we
  * call initializeEditor(). This is due to metaBox timing.
  *
- * @param {string}  id       Unique identifier for editor instance.
- * @param {Object}  post     API entity for post to edit.
- * @param {?Object} settings Editor settings object.
+ * @param {string}  id            Unique identifier for editor instance.
+ * @param {Object}  postId        ID of the post to edit.
+ * @param {Object}  postType      Post type of the post to edit.
+ * @param {?Object} settings      Editor settings object.
+ * @param {Object}  overridePost  Post properties to override.
  *
  * @return {Object} Editor interface.
  */
-export function initializeEditor( id, post, settings ) {
+export function initializeEditor( id, postId, postType, settings, overridePost ) {
```

Comment:
In a potential future where `postId` becomes optional, would we want to support calling this as simply...

```js
initializeEditor( 'gutenberg', 'post' );
```

i.e. with `postType` before the optional `postId`

Asides:

- `id` is pretty useless now. It was originally intended for separate instances of an editor, which we don't really support. Maybe it should just be dropped.
- At what point might we consider an object argument? Since we could have many variations (post with ID and no overrides, post with no ID and overrides, post with overrides but no settings)

Reply:
I'll keep the `id` for now. I'm not certain if we'll support multi-instances at some point but prefer to keep it for now.

An object argument: I don't know 🤷‍♂️, It may feel like a component though and maybe keeping it as clarify the distinction?